### PR TITLE
Allow pt-disabled colors in label, select and input-group, and update documentation

### LINKED ISSUE #402: Form Controls JS API: Labels for disabled Radio, Checkbox, and Switch not grayed
### Bug report

- __Package version(s)__: 1.4.0
- __Browser and OS versions__: All

#### Actual behavior

[Disabled Labels](http://blueprintjs.com/docs/#components.forms.label.disabled) can be set with the CSS API using the `pt-disabled` modifer. This grays the text. 

When using the JS API for Radio buttons, Checkboxes, and Switches, setting the component to disabled (e.g. `<Checkbox disabled />`) does not set the label to the disabled gray color.

ce1b1ddc-cea1-11e6-9c39-314c329c927f-1

#### Expected behavior

Expected behavior is that labels are greyed out when disabled via the JS API, like they are when using the CSS API (e.g. `<label class="pt-label pt-disabled">`).

screen shot 2017-01-03 at 4 21 55 pm

### PR DESCRIPTION
**Pull request by leebyp**
#### Checklist

- [x] Update documentation

#### Changes proposed in this pull request:

- Allow disabled label color with `.pt-disabled`.
- Allow input-group icon color with `.pt-disabled`.
- Allow select arrow color with `.pt-disabled`.
- Update CSS API documentation for `:disabled` to include `.pt-disabled` usage.

### CODE REVIEW

**Comment by giladgray** [comment]
this is not really a "fix" since you're implementing a new modifier.

File: packages/core/src/components/forms/controls.tsx [hunk 1]
```diff
@@ -39,10 +40,15 @@ export class Control<P extends IControlProps> extends React.Component<React.HTML
     // generates control markup for given input type.
     // optional inputRef in case the component needs reference for itself (don't forget to invoke the prop!).
     protected renderControl(type: "checkbox" | "radio", typeClassName: string, inputRef = this.props.inputRef) {
+        const className = classNames(
+            Classes.CONTROL,
+            typeClassName,
+            this.props.className,
```
Comment (giladgray) [inline packages/core/src/components/forms/controls.tsx]: `this.props.className` should always come last in `classNames()` so user classes take precedence.

File: packages/core/src/components/forms/controls.tsx [hunk 2]
```diff
@@ -39,10 +40,15 @@ export class Control<P extends IControlProps> extends React.Component<React.HTML
...
+            {[Classes.DISABLED]: this.props.disabled});
```
Comment (giladgray) [inline packages/core/src/components/forms/controls.tsx]: spaces inside braces. `{ [key]: val }`

**Comment by giladgray** [comment]
easy way to test: tick the "Inline" switch in Popover example.

image

File: packages/core/src/components/forms/_controls.scss [hunk 3]
```diff
@@ -169,7 +174,7 @@ $control-checked-background-color-active: nth(map-get($button-intents, "primary"
   </label>
 
   :checked  - Checked
-  :disabled - Disabled
+  :disabled - Disabled. This should be used with a <code>.pt-disabled</code> modifier (not shown below).
```
Comment (giladgray) [inline packages/core/src/components/forms/_controls.scss]: this doesn't explain _how_ to use it. say something about "add .pt-disabled to .pt-control element to change text color," and for all the similar notes below.

File: packages/core/src/components/forms/_input-group.scss [hunk 4]
```diff
@@ -51,7 +51,7 @@ Markup:
   <button class="pt-button pt-minimal pt-intent-primary pt-icon-arrow-right" {{:modifier}}></button>
 </div>
 
-:disabled - Disabled input. Must be added separately to the <code>&#60;input&#62;</code> and <code>&#60;button&#62;</code>.
+:disabled - Disabled input. Must be added separately to the <code>&#60;input&#62;</code> and <code>&#60;button&#62;</code>. Should be used with a <code>.pt-disabled</code> modifier (not shown below).
```
Comment (giladgray) [inline packages/core/src/components/forms/_input-group.scss]: "Also add `.pt-disabled` to `.pt-input-group` for icon coloring."

File: packages/core/src/components/forms/_input.scss [hunk 5]
```diff
@@ -76,7 +76,7 @@ Markup:
   <input class="pt-input" {{:modifier}} type="search" placeholder="Search input" dir="auto" />
 </div>
 
-:disabled - Disabled
+:disabled - Disabled. This should be used with a <code>.pt-disabled</code> modifier (not shown below).
```
Comment (giladgray) [inline packages/core/src/components/forms/_input.scss]: this shouldn't be necessary. `disabled` attribute is sufficient for `.pt-input`

File: packages/core/src/components/forms/_select.scss [hunk 6]
```diff
@@ -28,7 +28,7 @@ Markup:
   </select>
 </div>
 
-:disabled - Disabled
+:disabled - Disabled. This should be used with a <code>.pt-disabled</code> modifier (not shown below).
```
Comment (giladgray) [inline packages/core/src/components/forms/_select.scss]: "Add .pt-disabled to .pt-select for icon coloring."

File: packages/core/src/components/forms/_input.scss (hunk 5 above)
Comment (leebyp) [inline packages/core/src/components/forms/_input.scss]: This example includes an icon - reworded to be more obvious.

**Comment by leebyp** [comment]
updated PR title, PR description and documentation changes

File: packages/core/src/components/forms/controls.tsx [hunk 7]
```diff
@@ -39,10 +40,15 @@ export class Control<P extends IControlProps> extends React.Component<React.HTML
     // generates control markup for given input type.
     // optional inputRef in case the component needs reference for itself (don't forget to invoke the prop!).
     protected renderControl(type: "checkbox" | "radio", typeClassName: string, inputRef = this.props.inputRef) {
+        const className = classNames(
+            Classes.CONTROL,
+            typeClassName,
+            { [Classes.DISABLED]: this.props.disabled },
+            this.props.className);
```
Comment (giladgray) [inline packages/core/src/components/forms/controls.tsx]: close paren should be on its own line, dedented back to original.

```
function(
    args
);
```

File: packages/core/src/components/forms/_controls.scss [hunk 8]
```diff
@@ -343,7 +348,7 @@ $control-checked-background-color-active: nth(map-get($button-intents, "primary"
   </label>
 
   :checked  - Selected
-  :disabled - Disabled
+  :disabled - Disabled. Add <code>.pt-disabled</code> to <code>.pt-control</code> to change text color (not shown below).
```
Comment (giladgray) [inline packages/core/src/components/forms/_controls.scss]: "Also add..." to indicate that some additional action is suggested?

**Comment by leebyp** [comment]
Updated the `Disabled Labels` part of the docs

**Pull request merged**